istioctl/cmd/analyze.go

```diff
@@ -316,6 +317,9 @@ func Analyze() *cobra.Command {
 		"Process directory arguments recursively. Useful when you want to analyze related manifests organized within the same directory.")
 	analysisCmd.PersistentFlags().BoolVar(&ignoreUnknown, "ignore-unknown", false,
 		"Don't complain about un-parseable input documents, for cases where analyze should run only on k8s compliant inputs.")
+	analysisCmd.PersistentFlags().BoolVar(&allowAPIVersionConversion, "allowAPIVersionConversion", false,
```

Comment:
Why would someone ever *not* want this?

Reply:
I thought it would be issues like the one you mentioned below, and so just let users decide whether to use the feature or not. Since it will be version aliases, the conversion will be safe, and I'll remove the flag.